Assets/SEE/UI/PropertyDialog/UserPropertyDialog.cs

```diff
@@ -0,0 +1,162 @@
+        /// Event triggered when the user presses the OK button. Clients can
+        /// register on this event to receive a notification when this happens.
+        /// </summary>
+        public readonly UnityEvent OnConfirm = new();
+
+        /// <summary>
+        /// Event triggered when the user presses the Cancel button. Clients can
+        /// register on this event to receive a notification when this happens.
+        /// </summary>
+        public readonly UnityEvent OnCancel = new();
+
+
+
+        /// <summary>
+        /// The network configuration to be manipulated by this dialog.
+        /// </summary>
+        private Net.Network networkConfig;
+
+        /// <summary>
+        /// Creates and opens the dialog.
+        /// </summary>
+        public void Open()
+        {
+            networkConfig.Load();
+
+            dialog = new GameObject("User settings");
+            PropertyGroup group = dialog.AddComponent<PropertyGroup>();
+            group.Name = "User settings";
+            {
+                playerName = dialog.AddComponent<StringProperty>();
+                playerName.Name = "Username";
+                playerName.Value = networkConfig.Playername;
+                playerName.Description = "Username which will be used as avatar-tag and in other function.";
+                group.AddProperty(playerName);
+            }
+            // Dialog
+            propertyDialog = dialog.AddComponent<PropertyDialog>();
+            propertyDialog.Title = "User settings";
+            propertyDialog.Description = "Enter the user settings";
+            propertyDialog.AddGroup(group);
+
+            // Register listeners
+            propertyDialog.OnConfirm.AddListener(OKButtonPressed);
+            propertyDialog.OnCancel.AddListener(CancelButtonPressed);
+
+            SEEInput.KeyboardShortcutsEnabled = false;
+            // Go online
+            propertyDialog.DialogShouldBeShown = true;
+        }
+
+        /// <summary>
+        /// Notifies all listeners on <see cref="OnCancel"/> and closes the dialog.
+        /// </summary>
+        private void CancelButtonPressed()
+        {
+            propertyDialog.Close();
+            OnCancel.Invoke();
+            SEEInput.KeyboardShortcutsEnabled = true;
+            Close();
+        }
+
+        /// <summary>
+        /// Notifies all listeners on <see cref="OnConfirm"/> and save name and closes the dialog.
+        /// </summary>
+        private void OKButtonPressed()
+        {
+            bool errorOccurred = false;
+            {
+                string playername = playerName.Value.Trim();
+
+                if (ValidUsername(playername))
+                {
+                    networkConfig.Playername = playername;
+                }
+                else
+                {
+                    ShowNotification.Error
+                       ("Username Error",
+                        "Username needs to be minimum one character long.");
```

Comment:
The formatting here is weird (e.g., there shouldn't be a newline before calling a function).

Reply:
copied from NetworkPropertyDialog and adapt it, but will change same both ;-)